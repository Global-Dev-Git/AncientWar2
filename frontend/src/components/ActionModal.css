.action-modal__backdrop {
  position: fixed;
  inset: 0;
  background: rgba(10, 16, 24, 0.6);
  display: grid;
  place-items: center;
  z-index: 200;
  padding: 2rem;
}

.action-modal {
  width: min(480px, 100%);
  background: var(--surface-panel);
  border-radius: 1.25rem;
  box-shadow: var(--shadow-soft);
  padding: 1.5rem;
  display: flex;
  flex-direction: column;
  gap: 1rem;
  animation: modal-pop 0.3s ease;
}

.action-modal header {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
}

.action-modal header h3 {
  margin: 0;
  color: var(--text-strong);
}

.action-modal header button {
  border: none;
  background: transparent;
  color: var(--text-muted);
  font-size: 1.2rem;
}

.action-modal__description {
  margin: 0;
  font-size: 0.95rem;
  color: var(--text-muted);
  line-height: 1.5;
}

.action-modal__field {
  display: flex;
  flex-direction: column;
  gap: 0.35rem;
  font-size: 0.85rem;
  color: var(--text-muted);
}

.action-modal__field select {
  padding: 0.6rem 0.75rem;
  border-radius: 0.75rem;
  border: 1px solid rgba(118, 138, 166, 0.4);
  background: var(--surface-elevated);
  color: var(--text-strong);
}

.action-modal footer {
  display: flex;
  justify-content: flex-end;
  gap: 0.75rem;
}

.action-modal footer button {
  padding: 0.65rem 1.2rem;
  border-radius: 0.85rem;
  border: none;
  font-weight: 600;
  cursor: pointer;
  transition: transform 0.15s ease, box-shadow 0.15s ease;
}

.action-modal footer button.secondary {
  background: var(--surface-elevated);
  color: var(--text-strong);
}

.action-modal footer button:not(.secondary) {
  background: linear-gradient(135deg, #5886ff, #3b54e2);
  color: white;
  box-shadow: 0 12px 24px rgba(88, 134, 255, 0.25);
}

.action-modal footer button:disabled {
  opacity: 0.5;
  cursor: not-allowed;
  box-shadow: none;
}

@keyframes modal-pop {
  from {
    opacity: 0;
    transform: translateY(12px) scale(0.95);
  }
  to {
    opacity: 1;
    transform: translateY(0) scale(1);
  }
}
