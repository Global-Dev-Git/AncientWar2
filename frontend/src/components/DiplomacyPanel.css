.diplomacy-panel {
  background: var(--surface-panel);
  border-radius: 1.25rem;
  padding: 1.25rem;
  box-shadow: var(--shadow-subtle);
  display: flex;
  flex-direction: column;
  gap: 1rem;
  height: 100%;
}

.diplomacy-panel h3 {
  margin: 0;
  text-transform: uppercase;
  letter-spacing: 0.12em;
  font-size: 0.85rem;
  color: var(--text-muted);
}

.diplomacy-panel table {
  width: 100%;
  border-collapse: collapse;
  font-size: 0.8rem;
}

.diplomacy-panel th,
.diplomacy-panel td {
  padding: 0.6rem;
  text-align: left;
  border-bottom: 1px solid rgba(255, 255, 255, 0.05);
}

.diplomacy-panel th {
  font-weight: 600;
  color: var(--text-muted);
  text-transform: uppercase;
  letter-spacing: 0.08em;
}

.diplomacy-row {
  background: var(--surface-elevated);
  border-left: 3px solid transparent;
}

.diplomacy-row--ally {
  border-left-color: rgba(72, 199, 116, 0.55);
}

.diplomacy-row--hostile {
  border-left-color: rgba(255, 99, 71, 0.55);
}

.diplomacy-row__nation {
  display: flex;
  flex-direction: column;
  gap: 0.25rem;
}

.diplomacy-row__nation strong {
  color: var(--text-strong);
}

.diplomacy-row__nation span {
  color: var(--text-muted);
  font-size: 0.75rem;
}

.diplomacy-score {
  font-weight: 700;
}

.diplomacy-row__badges {
  display: flex;
  gap: 0.25rem;
}

.tag {
  font-size: 0.7rem;
  padding: 0.25rem 0.5rem;
  border-radius: 0.6rem;
  letter-spacing: 0.05em;
  text-transform: uppercase;
}

.tag--war {
  background: rgba(255, 99, 71, 0.15);
  color: #ff6347;
}

.tag--ally {
  background: rgba(72, 199, 116, 0.15);
  color: #36b56e;
}
