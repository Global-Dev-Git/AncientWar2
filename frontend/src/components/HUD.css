.hud {
  display: flex;
  flex-direction: column;
  gap: 1.25rem;
  background: var(--surface-panel);
  border-radius: 1.5rem;
  padding: 1.5rem;
  box-shadow: var(--shadow-soft);
}

.hud__header {
  display: flex;
  justify-content: space-between;
  gap: 1.5rem;
  align-items: flex-start;
}

.hud__header h1 {
  margin: 0;
  font-size: 1.8rem;
  color: var(--text-strong);
}

.hud__header p {
  margin: 0.35rem 0 0;
  color: var(--text-muted);
  line-height: 1.6;
}

.hud__meta {
  display: flex;
  gap: 0.75rem;
  font-size: 0.85rem;
  background: var(--surface-elevated);
  padding: 0.75rem 1rem;
  border-radius: 1rem;
  align-items: center;
  color: var(--text-muted);
}

.hud__stats {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(180px, 1fr));
  gap: 0.75rem 1rem;
}

.hud__economy {
  background: var(--surface-elevated);
  border-radius: 1.25rem;
  padding: 1rem 1.25rem;
  display: flex;
  flex-direction: column;
  gap: 1rem;
}

.hud__economy-header {
  display: flex;
  justify-content: space-between;
  gap: 1rem;
  align-items: center;
}

.hud__economy-header h2 {
  margin: 0;
  font-size: 1.2rem;
  color: var(--text-strong);
}

.hud__economy-metrics {
  display: flex;
  flex-wrap: wrap;
  gap: 0.75rem;
  font-size: 0.85rem;
  color: var(--text-muted);
}

.hud__economy-grid {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(140px, 1fr));
  gap: 0.75rem;
}

.hud__economy-card {
  background: rgba(12, 18, 30, 0.07);
  border-radius: 0.9rem;
  padding: 0.75rem;
  display: flex;
  flex-direction: column;
  gap: 0.4rem;
}

.hud__economy-card h3 {
  display: flex;
  justify-content: space-between;
  align-items: baseline;
  margin: 0;
  font-size: 0.95rem;
  color: var(--text-strong);
}

.hud__economy-price {
  font-size: 0.9rem;
  color: var(--text-muted);
  font-weight: 600;
}

.hud__economy-history {
  margin: 0;
  font-size: 0.7rem;
  letter-spacing: 0.04em;
  color: var(--text-muted);
}
