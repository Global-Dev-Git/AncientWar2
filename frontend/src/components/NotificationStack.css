.notification-stack {
  position: fixed;
  right: 2rem;
  bottom: 2rem;
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  z-index: 100;
}

.notification {
  padding: 0.75rem 1rem;
  border-radius: 0.75rem;
  backdrop-filter: blur(12px);
  box-shadow: var(--shadow-soft);
  font-size: 0.9rem;
  max-width: 18rem;
  animation: notification-pop 0.4s ease;
}

.notification--positive {
  background: linear-gradient(135deg, rgba(72, 199, 116, 0.8), rgba(48, 146, 78, 0.8));
  color: #0b2e16;
}

.notification--negative {
  background: linear-gradient(135deg, rgba(255, 99, 71, 0.85), rgba(168, 32, 32, 0.85));
  color: #fff7f7;
}

.notification--neutral {
  background: linear-gradient(135deg, rgba(118, 138, 166, 0.8), rgba(68, 84, 105, 0.8));
  color: #f5f8ff;
}

@keyframes notification-pop {
  from {
    opacity: 0;
    transform: translateY(12px) scale(0.95);
  }
  to {
    opacity: 1;
    transform: translateY(0) scale(1);
  }
}
