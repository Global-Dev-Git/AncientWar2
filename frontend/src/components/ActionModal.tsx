import { useEffect, useMemo, useState } from 'react'
import type { ActionType, NationState, PlayerAction, TerritoryState } from '../game/types'
import { ACTION_LABELS } from '../game/constants'
import './ActionModal.css'

interface ActionModalProps {
  actionType: ActionType | null
  onClose: () => void
  onConfirm: (action: PlayerAction) => void
  nations: NationState[]
  territories: TerritoryState[]
  playerNationId: string
  selectedTerritoryId?: string
}

const needsTargetNation: ActionType[] = ['Spy', 'DiplomacyOffer', 'DeclareWar', 'FormAlliance', 'Bribe']
const needsSourceTerritory: ActionType[] = ['RecruitArmy', 'MoveArmy']

const describeEffect = (action: ActionType): string => {
  switch (action) {
    case 'InvestInTech':
      return 'Spend coin to gain technology and science.'
    case 'RecruitArmy':
      return 'Raise garrison strength in the selected territory.'
    case 'MoveArmy':
      return 'Shift troops to friendly lands or attack adjacent enemies.'
    case 'CollectTaxes':
      return 'Gain treasury income, but crime will rise.'
    case 'PassLaw':
      return 'Improve laws and stability, with cultural side effects.'
    case 'Spy':
      return 'Destabilise a rival and raise their crime.'
    case 'DiplomacyOffer':
      return 'Improve relations; special bonuses for maritime deals.'
    case 'DeclareWar':
      return 'Begin open conflict. Stability will drop.'
    case 'FormAlliance':
      return 'Form a mutual defense pact and boost relations.'
    case 'Bribe':
      return 'Buy influence at the cost of coin and their integrity.'
    case 'SuppressCrime':
      return 'Reduce crime but lose a little support.'
    default:
      return ''
  }
}

export const ActionModal = ({
  actionType,
  onClose,
  onConfirm,
  nations,
  territories,
  playerNationId,
  selectedTerritoryId,
}: ActionModalProps) => {
  const [targetNationId, setTargetNationId] = useState<string>('')
  const [sourceTerritoryId, setSourceTerritoryId] = useState<string | undefined>(selectedTerritoryId)
  const [targetTerritoryId, setTargetTerritoryId] = useState<string>('')

  useEffect(() => {
    setTargetNationId('')
    setTargetTerritoryId('')
    setSourceTerritoryId(selectedTerritoryId)
  }, [actionType, selectedTerritoryId])

  const availableNationTargets = useMemo(
    () => nations.filter((nation) => nation.id !== playerNationId),
    [nations, playerNationId],
  )

  const controlledTerritories = useMemo(
    () => territories.filter((territory) => territory.ownerId === playerNationId),
    [territories, playerNationId],
  )

  const neighborOptions = useMemo(() => {
    if (!sourceTerritoryId) return []
    const source = territories.find((territory) => territory.id === sourceTerritoryId)
    if (!source) return []
    return source.neighbors.map((id) => territories.find((territory) => territory.id === id)).filter(Boolean) as TerritoryState[]
  }, [sourceTerritoryId, territories])

  if (!actionType) return null

  const requiresNation = needsTargetNation.includes(actionType)
  const requiresSource = needsSourceTerritory.includes(actionType)
  const requiresTargetTerritory = actionType === 'MoveArmy'

  const canConfirm = () => {
    if (requiresNation && !targetNationId) return false
    if (requiresSource && !sourceTerritoryId) return false
    if (requiresTargetTerritory && !targetTerritoryId) return false
    return true
  }

  const confirm = () => {
    if (!canConfirm()) return
    const payload: PlayerAction = {
      type: actionType,
      targetNationId: targetNationId || undefined,
      sourceTerritoryId,
      targetTerritoryId: targetTerritoryId || undefined,
    }
    onConfirm(payload)
    onClose()
  }

  return (
    <div className="action-modal__backdrop" role="dialog" aria-modal>
      <div className="action-modal">
        <header>
          <h3>{ACTION_LABELS[actionType]}</h3>
          <button type="button" onClick={onClose} aria-label="Close action dialog">
            ✕
          </button>
        </header>
        <p className="action-modal__description">{describeEffect(actionType)}</p>

        {requiresSource && (
          <label className="action-modal__field">
            <span>Source Territory</span>
            <select
              value={sourceTerritoryId ?? ''}
              onChange={(event) => setSourceTerritoryId(event.target.value || undefined)}
            >
              <option value="">Select territory</option>
              {controlledTerritories.map((territory) => (
                <option key={territory.id} value={territory.id}>
                  {territory.name}
                </option>
              ))}
            </select>
          </label>
        )}

        {requiresTargetTerritory && (
          <label className="action-modal__field">
            <span>Target Territory</span>
            <select value={targetTerritoryId} onChange={(event) => setTargetTerritoryId(event.target.value)}>
              <option value="">Select target</option>
              {neighborOptions.map((territory) => (
                <option key={territory.id} value={territory.id}>
                  {territory.name} — {territory.ownerId === playerNationId ? 'Friendly' : 'Enemy'}
                </option>
              ))}
            </select>
          </label>
        )}

        {requiresNation && (
          <label className="action-modal__field">
            <span>Target Nation</span>
            <select value={targetNationId} onChange={(event) => setTargetNationId(event.target.value)}>
              <option value="">Choose nation</option>
              {availableNationTargets.map((nation) => (
                <option key={nation.id} value={nation.id}>
                  {nation.name}
                </option>
              ))}
            </select>
          </label>
        )}

        <footer>
          <button type="button" className="secondary" onClick={onClose}>
            Cancel
          </button>
          <button type="button" onClick={confirm} disabled={!canConfirm()}>
            Confirm
          </button>
        </footer>
      </div>
    </div>
  )
}

export default ActionModal
