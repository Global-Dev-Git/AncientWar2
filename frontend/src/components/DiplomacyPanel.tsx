import type { DiplomacyMatrix, NationState } from '../game/types'
import './DiplomacyPanel.css'

interface DiplomacyPanelProps {
  playerNation: NationState
  nations: NationState[]
  diplomacy: DiplomacyMatrix
}

const relationTone = (score: number): 'ally' | 'neutral' | 'hostile' => {
  if (score >= 40) return 'ally'
  if (score <= -20) return 'hostile'
  return 'neutral'
}

const formatTreaties = (treaties: string[]): string => (treaties.length ? treaties.join(', ') : 'None')

export const DiplomacyPanel = ({ playerNation, nations, diplomacy }: DiplomacyPanelProps) => (
  <div className="diplomacy-panel">
    <h3>Diplomacy</h3>
    <table>
      <thead>
        <tr>
          <th>Nation</th>
          <th>Relation</th>
          <th>Reputation</th>
          <th>Treaties</th>
          <th>Casus Belli</th>
          <th>Penalties</th>
          <th>Status</th>
        </tr>
      </thead>
      <tbody>
        {nations
          .filter((nation) => nation.id !== playerNation.id)
          .map((nation) => {
            const status =
              diplomacy.relations[playerNation.id]?.[nation.id] ??
              {
                score: 0,
                treaties: [],
                casusBelli: null,
                treatyPenalty: 0,
              }
            const tone = relationTone(status.score)
            const key = [playerNation.id, nation.id].sort().join('|')
            const atWar = diplomacy.wars.has(key)
            const allied = diplomacy.alliances.has(key)
            const reputation = diplomacy.reputations[nation.id] ?? 0
            return (
              <tr key={nation.id} className={`diplomacy-row diplomacy-row--${tone}`}>
                <td>
                  <div className="diplomacy-row__nation">
                    <strong>{nation.name}</strong>
                    <span>{nation.description}</span>
                  </div>
                </td>
                <td className="diplomacy-score">{status.score}</td>
                <td>{reputation}</td>
                <td>{formatTreaties(status.treaties)}</td>
                <td>{status.casusBelli ?? 'None'}</td>
                <td>{status.treatyPenalty}</td>
                <td className="diplomacy-row__badges">
                  {atWar && <span className="tag tag--war">War</span>}
                  {allied && <span className="tag tag--ally">Alliance</span>}
                </td>
              </tr>
            )
          })}
      </tbody>
    </table>
  </div>
)

export default DiplomacyPanel
