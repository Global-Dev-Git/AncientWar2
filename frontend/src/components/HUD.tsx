import type { NationState } from '../game/types'
import StatBar from './StatBar'
import './HUD.css'

interface HUDProps {
  nation: NationState
  treasury: number
  turn: number
  actionsRemaining: number
}

const statConfig: Array<{ key: keyof NationState['stats']; label: string; invert?: boolean }> = [
  { key: 'stability', label: 'Stability' },
  { key: 'military', label: 'Military' },
  { key: 'tech', label: 'Technology' },
  { key: 'economy', label: 'Economy' },
  { key: 'crime', label: 'Crime', invert: true },
  { key: 'influence', label: 'Influence' },
  { key: 'support', label: 'Support' },
  { key: 'science', label: 'Science' },
  { key: 'laws', label: 'Laws' },
]

const toneFor = (value: number, invert?: boolean) => {
  if (invert) {
    if (value <= 25) return 'stable'
    if (value <= 60) return 'risky'
    return 'critical'
  }
  if (value >= 70) return 'stable'
  if (value >= 40) return 'risky'
  return 'critical'
}

export const HUD = ({ nation, treasury, turn, actionsRemaining }: HUDProps) => (
  <section className="hud">
    <header className="hud__header">
      <div>
        <h1>{nation.name}</h1>
        <p>{nation.description}</p>
      </div>
      <div className="hud__meta">
        <span>Turn {turn}</span>
        <span>Treasury {treasury}</span>
        <span>Actions left {actionsRemaining}</span>
      </div>
    </header>
    <div className="hud__stats">
      {statConfig.map((stat) => (
        <StatBar
          key={stat.key}
          label={stat.label}
          value={nation.stats[stat.key]}
          tone={toneFor(nation.stats[stat.key], stat.invert)}
        />
      ))}
    </div>
  </section>
)

export default HUD
