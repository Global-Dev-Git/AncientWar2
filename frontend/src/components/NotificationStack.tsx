import type { NotificationEntry } from '../game/types'
import './NotificationStack.css'

interface NotificationStackProps {
  notifications: NotificationEntry[]
}

export const NotificationStack = ({ notifications }: NotificationStackProps) => (
  <div className="notification-stack">
    {notifications.map((notification) => (
      <div key={notification.id} className={`notification notification--${notification.tone}`}>
        {notification.message}
      </div>
    ))}
  </div>
)

export default NotificationStack
