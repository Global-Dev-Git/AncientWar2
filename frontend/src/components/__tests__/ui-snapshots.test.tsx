import { render } from '@testing-library/react'
import { describe, expect, it } from 'vitest'
import ActionModal from '../ActionModal'
import DiplomacyPanel from '../DiplomacyPanel'
import HUD from '../HUD'
import NotificationStack from '../NotificationStack'
import StatBar from '../StatBar'
import type { DiplomacyMatrix, NationState, NotificationEntry, TerritoryState } from '../../game/types'

const sampleNation = (): NationState => ({
  id: 'rome',
  name: 'Roman Republic',
  description: 'Legions prepared for campaigns across the Mediterranean.',
  territories: ['rome_latium', 'rome_etruria'],
  stats: {
    stability: 72,
    military: 85,
    tech: 63,
    economy: 58,
    crime: 22,
    influence: 48,
    support: 71,
    science: 54,
    laws: 60,
  },
  advantages: ['Disciplined Legions'],
  disadvantages: ['Senatorial Intrigue'],
  armies: [
    { id: 'rome-army-1', territoryId: 'rome_latium', strength: 7 },
    { id: 'rome-army-2', territoryId: 'rome_etruria', strength: 5 },
  ],
  treasury: 90,
})

const rivalNation = (): NationState => ({
  id: 'carthage',
  name: 'Carthage',
  description: 'Maritime traders contesting sea lanes.',
  territories: ['carthage_carthage'],
  stats: {
    stability: 65,
    military: 70,
    tech: 59,
    economy: 80,
    crime: 28,
    influence: 62,
    support: 60,
    science: 50,
    laws: 55,
  },
  advantages: ['Naval Supremacy'],
  disadvantages: ['Mercenary Reliance'],
  armies: [{ id: 'carthage-army-1', territoryId: 'carthage_carthage', strength: 6 }],
  treasury: 85,
})

const sampleTerritories = (): TerritoryState[] => [
  {
    id: 'rome_latium',
    name: 'Latium',
    coordinates: [0, 0],
    terrain: 'plains',
    neighbors: ['rome_etruria', 'carthage_carthage'],
    ownerId: 'rome',
    garrison: 7,
    development: 60,
    unrest: 10,
  },
  {
    id: 'rome_etruria',
    name: 'Etruria',
    coordinates: [1, 0],
    terrain: 'hills',
    neighbors: ['rome_latium', 'carthage_carthage'],
    ownerId: 'rome',
    garrison: 5,
    development: 55,
    unrest: 12,
  },
  {
    id: 'carthage_carthage',
    name: 'Carthage',
    coordinates: [2, 1],
    terrain: 'coastal',
    neighbors: ['rome_latium', 'rome_etruria'],
    ownerId: 'carthage',
    garrison: 6,
    development: 65,
    unrest: 18,
  },
]

const sampleDiplomacy = (): DiplomacyMatrix => ({
  relations: {
    rome: { carthage: -35 },
    carthage: { rome: -35 },
  },
  wars: new Set(['carthage|rome']),
  alliances: new Set<string>(),
  blockades: { 'carthage|rome': 0.4 },
})

const sampleNotifications = (): NotificationEntry[] => [
  { id: '1', message: 'Latium celebrates a festival.', tone: 'positive', timestamp: 1 },
  { id: '2', message: 'Carthaginian corsairs harass traders.', tone: 'negative', timestamp: 2 },
]

describe('UI snapshot coverage', () => {
  it('renders stat bar snapshot', () => {
    const { asFragment } = render(<StatBar label="Military" value={85} tone="stable" />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders HUD stat panel snapshot', () => {
    const nation = sampleNation()
    const { asFragment } = render(<HUD nation={nation} treasury={nation.treasury} turn={5} actionsRemaining={2} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders diplomacy treaty rows snapshot', () => {
    const { asFragment } = render(
      <DiplomacyPanel
        playerNation={sampleNation()}
        nations={[sampleNation(), rivalNation()]}
        diplomacy={sampleDiplomacy()}
      />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders army inspector modal snapshot', () => {
    const territories = sampleTerritories()
    const { asFragment } = render(
      <ActionModal
        actionType="MoveArmy"
        onClose={() => undefined}
        onConfirm={() => undefined}
        nations={[sampleNation(), rivalNation()]}
        territories={territories}
        playerNationId="rome"
        selectedTerritoryId="rome_latium"
      />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders notification panel snapshot', () => {
    const { asFragment } = render(<NotificationStack notifications={sampleNotifications()} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
