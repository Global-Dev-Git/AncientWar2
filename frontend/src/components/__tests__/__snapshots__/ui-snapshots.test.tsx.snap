// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UI snapshot coverage > renders HUD stat panel snapshot 1`] = `
<DocumentFragment>
  <section
    class="hud"
  >
    <header
      class="hud__header"
    >
      <div>
        <h1>
          Roman Republic
        </h1>
        <p>
          Legions prepared for campaigns across the Mediterranean.
        </p>
      </div>
      <div
        class="hud__meta"
      >
        <span>
          Turn 5
        </span>
        <span>
          Treasury 90
        </span>
        <span>
          Actions left 2
        </span>
      </div>
    </header>
    <div
      class="hud__stats"
    >
      <div
        class="stat-bar"
      >
        <div
          class="stat-bar__label"
        >
          <span>
            Stability
          </span>
          <strong>
            72
          </strong>
        </div>
        <div
          class="stat-bar__meter"
        >
          <div
            class="stat-bar__fill"
            style="width: 72%; background: var(--accent-positive);"
          />
        </div>
      </div>
      <div
        class="stat-bar"
      >
        <div
          class="stat-bar__label"
        >
          <span>
            Military
          </span>
          <strong>
            85
          </strong>
        </div>
        <div
          class="stat-bar__meter"
        >
          <div
            class="stat-bar__fill"
            style="width: 85%; background: var(--accent-positive);"
          />
        </div>
      </div>
      <div
        class="stat-bar"
      >
        <div
          class="stat-bar__label"
        >
          <span>
            Technology
          </span>
          <strong>
            63
          </strong>
        </div>
        <div
          class="stat-bar__meter"
        >
          <div
            class="stat-bar__fill"
            style="width: 63%; background: var(--accent-warning);"
          />
        </div>
      </div>
      <div
        class="stat-bar"
      >
        <div
          class="stat-bar__label"
        >
          <span>
            Economy
          </span>
          <strong>
            58
          </strong>
        </div>
        <div
          class="stat-bar__meter"
        >
          <div
            class="stat-bar__fill"
            style="width: 58%; background: var(--accent-warning);"
          />
        </div>
      </div>
      <div
        class="stat-bar"
      >
        <div
          class="stat-bar__label"
        >
          <span>
            Crime
          </span>
          <strong>
            22
          </strong>
        </div>
        <div
          class="stat-bar__meter"
        >
          <div
            class="stat-bar__fill"
            style="width: 22%; background: var(--accent-positive);"
          />
        </div>
      </div>
      <div
        class="stat-bar"
      >
        <div
          class="stat-bar__label"
        >
          <span>
            Influence
          </span>
          <strong>
            48
          </strong>
        </div>
        <div
          class="stat-bar__meter"
        >
          <div
            class="stat-bar__fill"
            style="width: 48%; background: var(--accent-warning);"
          />
        </div>
      </div>
      <div
        class="stat-bar"
      >
        <div
          class="stat-bar__label"
        >
          <span>
            Support
          </span>
          <strong>
            71
          </strong>
        </div>
        <div
          class="stat-bar__meter"
        >
          <div
            class="stat-bar__fill"
            style="width: 71%; background: var(--accent-positive);"
          />
        </div>
      </div>
      <div
        class="stat-bar"
      >
        <div
          class="stat-bar__label"
        >
          <span>
            Science
          </span>
          <strong>
            54
          </strong>
        </div>
        <div
          class="stat-bar__meter"
        >
          <div
            class="stat-bar__fill"
            style="width: 54%; background: var(--accent-warning);"
          />
        </div>
      </div>
      <div
        class="stat-bar"
      >
        <div
          class="stat-bar__label"
        >
          <span>
            Laws
          </span>
          <strong>
            60
          </strong>
        </div>
        <div
          class="stat-bar__meter"
        >
          <div
            class="stat-bar__fill"
            style="width: 60%; background: var(--accent-warning);"
          />
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`UI snapshot coverage > renders army inspector modal snapshot 1`] = `
<DocumentFragment>
  <div
    aria-modal="true"
    class="action-modal__backdrop"
    role="dialog"
  >
    <div
      class="action-modal"
    >
      <header>
        <h3>
          Move Army
        </h3>
        <button
          aria-label="Close action dialog"
          type="button"
        >
          ✕
        </button>
      </header>
      <p
        class="action-modal__description"
      >
        Shift troops to friendly lands or attack adjacent enemies.
      </p>
      <label
        class="action-modal__field"
      >
        <span>
          Source Territory
        </span>
        bound HTMLSelectElement {
          "0": <option
            value=""
          >
            Select territory
          </option>,
          "1": <option
            value="rome_latium"
          >
            Latium
          </option>,
          "2": <option
            value="rome_etruria"
          >
            Etruria
          </option>,
        }
      </label>
      <label
        class="action-modal__field"
      >
        <span>
          Target Territory
        </span>
        bound HTMLSelectElement {
          "0": <option
            value=""
          >
            Select target
          </option>,
          "1": <option
            value="rome_etruria"
          >
            Etruria — Friendly
          </option>,
          "2": <option
            value="carthage_carthage"
          >
            Carthage — Enemy
          </option>,
        }
      </label>
      <footer>
        <button
          class="secondary"
          type="button"
        >
          Cancel
        </button>
        <button
          disabled=""
          type="button"
        >
          Confirm
        </button>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;

exports[`UI snapshot coverage > renders diplomacy treaty rows snapshot 1`] = `
<DocumentFragment>
  <div
    class="diplomacy-panel"
  >
    <h3>
      Diplomacy
    </h3>
    <ul>
      <li
        class="diplomacy-row diplomacy-row--hostile"
      >
        <div>
          <strong>
            Carthage
          </strong>
          <span>
            Maritime traders contesting sea lanes.
          </span>
        </div>
        <div
          class="diplomacy-row__status"
        >
          <span
            class="diplomacy-score"
          >
            -35
          </span>
          <span
            class="tag tag--war"
          >
            War
          </span>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`UI snapshot coverage > renders notification panel snapshot 1`] = `
<DocumentFragment>
  <div
    class="notification-stack"
  >
    <div
      class="notification notification--positive"
    >
      Latium celebrates a festival.
    </div>
    <div
      class="notification notification--negative"
    >
      Carthaginian corsairs harass traders.
    </div>
  </div>
</DocumentFragment>
`;

exports[`UI snapshot coverage > renders stat bar snapshot 1`] = `
<DocumentFragment>
  <div
    class="stat-bar"
  >
    <div
      class="stat-bar__label"
    >
      <span>
        Military
      </span>
      <strong>
        85
      </strong>
    </div>
    <div
      class="stat-bar__meter"
    >
      <div
        class="stat-bar__fill"
        style="width: 85%; background: var(--accent-positive);"
      />
    </div>
  </div>
</DocumentFragment>
`;
